src/shuffle.js

```diff
@@ -0,0 +1,22 @@
+'use strict'
+
+/**
+ * Performs pseudo random shuffles on clone of the array
+ *
+ * @param {Array<any>} array
+ * @returns {Array<any>}
+ */
+const shuffle = (array) => {
+  const clone = [...array]
+  const length = clone.length
+  for (let i = 0; i < length; i++) {
+    const mov = Math.floor(Math.random() * length)
```

Comment:
I think this shuffle function may not provide an even distribution of permutations, as you're always picking from the entire range of the array for every position.

May I suggest to this approach: 

´´´js
...
for (let i = length - 1; i > 0; i--) {
  const index = Math.floor(Math.random() * (i + 1)); // random index from 0 to i
  [clone[i], clone[index]] = [clone[index], clone[i]]; // swap elements
}
...
´´´

Reply:
Richard Durstenfeld algorithm?